projects/firmware_102/src/main.c

```diff
@@ -0,0 +1,44 @@
+#include <stdbool.h>
+#include <stdint.h>  // for integer types
+#include <stdlib.h>  // for random numbers
+#include "interrupt.h"
+#include "log.h"         // for printing
+#include "soft_timer.h"  // for soft timers
+#include "wait.h"        // for wait function
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void prv_timer_callback(SoftTimerId timer_id, void *context) {
+  Counters *count = context;
+  count->counter_a += 1;
+  LOG_DEBUG("Counter A:%d\n", count->counter_a);
+
+  if (count->counter_a % 2 == 0 && count->counter_a != 0) {
```

Comment:
I don't think you need to check count->counter_a != 0 since you increment counter_a before this line. This isn't a huge deal but just a bit redundant.